src/main/kotlin/lotto/domain/AutoLottoGenerator.kt

```diff
@@ -0,0 +1,14 @@
+package lotto.domain
+
+class AutoLottoGenerator : LottoGenerator {
+
+    override fun generate(amount: Int): List<Lotto> {
+        val result = mutableListOf<Lotto>()
+        repeat(amount) {
+            val randomNumbers = (LottoNumber.LOWER_LIMIT..LottoNumber.UPPER_LIMIT).toList().shuffled().take(Lotto.NUM_OF_LOTTO)
+            val lottoNumbers = randomNumbers.map { LottoNumber(it) }
+            result += Lotto(lottoNumbers)
+        }
+        return result
+    }
```

Comment:
함수 또한 작은 단위로 나누어주면 가독성있는 코드가 될 것 같아요🙂

```suggestion
    fun generate(amount: Int): List<Lotto> = List(amount) {
            Lotto(generateLottoNumbers())
        }

    private fun generateLottoNumbers(): List<LottoNumber> = (LottoNumber.LOWER_LIMIT..LottoNumber.UPPER_LIMIT).shuffled()
        .take(Lotto.NUM_OF_LOTTO)
        .map { LottoNumber(it) }
```